File: junyoung/나태지옥 4-2/String 삼성.py
# 체감 난이도 1/10, 파이썬은 슬라이싱으로 간단히 풀림

def solve():
    t = int(input())
    find_s = input()
    len_s = len(find_s)
    s = input()
    
    cnt = 0
    
    for i in range(len(s)-len_s+1):
        if s[i:i+len_s] == find_s:
            cnt += 1
            
    print('#{} {}'.format(t, cnt))
    
for t in range(10):
    solve()

File: junyoung/나태지옥 4-2/String 삼성.cpp
#include <iostream>
#include <string>

using namespace std;

void solve() {
    int t;
    cin >> t;

    string find_s;
    cin >> find_s;
    int len_s = find_s.length();
    
    string s;
    cin >> s;
    int len_s_total = s.length();

    int cnt = 0;

    for (int i = 0; i <= len_s_total - len_s; ++i) {
        if (s.substr(i, len_s) == find_s) { // substr 은 i부터 len_s만큼의 문자열을 반환
            cnt++;
        }
    }

    cout << "#" << t << " " << cnt << endl;
}

int main() {
    for (int t = 1; t <= 10; ++t) {
        solve();
    }
    return 0;
}
